src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotificationDefendantOfHearingHandler.java

```diff
@@ -53,27 +61,39 @@ protected Map<String, Callback> callbacks() {
 
     @Override
     public String camundaActivityId(CallbackParams callbackParams) {
-        return isEvent(callbackParams, NOTIFY_DEFENDANT1_HEARING) ? TASK_ID_DEFENDANT1
-            : TASK_ID_DEFENDANT2;
+        CaseEvent caseEvent = CaseEvent.valueOf(callbackParams.getRequest().getEventId());
+        if (NOTIFY_DEFENDANT1_HEARING.equals(caseEvent)) {
+            return TASK_ID_DEFENDANT1;
+        } else if (NOTIFY_DEFENDANT2_HEARING.equals(caseEvent)) {
+            return TASK_ID_DEFENDANT2;
+        } else if (NOTIFY_DEFENDANT1_HEARING_HMC.equals(caseEvent)) {
+            return TASK_ID_DEFENDANT1_HMC;
+        } else if (NOTIFY_DEFENDANT2_HEARING_HMC.equals(caseEvent)) {
+            return TASK_ID_DEFENDANT2_HMC;
+        } else {
+            throw new CallbackException(String.format(EVENT_NOT_FOUND_MESSAGE, caseEvent));
+        }
     }
 
     private CallbackResponse notifyDefendantHearing(CallbackParams callbackParams) {
         CaseData caseData = callbackParams.getCaseData();
 
-        //ToDo: Replace with AHN logic
-        if (isEvent(callbackParams, NOTIFY_DEFENDANT1_HEARING_HMC) || isEvent(callbackParams, NOTIFY_DEFENDANT2_HEARING_HMC)) {
-            return AboutToStartOrSubmitCallbackResponse.builder().build();
-        }
-
         String recipient = caseData.getRespondentSolicitor1EmailAddress();
 
         if (isEvent(callbackParams, NOTIFY_DEFENDANT1_HEARING)) {
             sendEmail(caseData, recipient, true);
-        } else {
+        } else if (isEvent(callbackParams, NOTIFY_DEFENDANT2_HEARING)) {
             if (nonNull(caseData.getRespondentSolicitor2EmailAddress())) {
                 recipient = caseData.getRespondentSolicitor2EmailAddress();
             }
             sendEmail(caseData, recipient, false);
+        } else if (isEvent(callbackParams, NOTIFY_DEFENDANT1_HEARING_HMC)) {
+            sendEmailHmc(caseData, recipient, true);
+        } else if (isEvent(callbackParams, NOTIFY_DEFENDANT2_HEARING_HMC)) {
+            if (nonNull(caseData.getRespondentSolicitor2EmailAddress())) {
+                recipient = caseData.getRespondentSolicitor2EmailAddress();
+            }
+            sendEmailHmc(caseData, recipient, false);
```

Comment:
this can be moved into the previous if statement

Reply:
updated